### ISSUE TITLE
Additional Markup tags

### ISSUE THREAD

**Issue by Amuri-Bonface** [eerotal/LibreSignage#83]
First of all excellent work.
I was wondering if there is a way to modify or add more tags that support features such as:
    -Clock
    -Scrolling text
    - Live RSS feeds
    -Twitter, facebook. Instagram and youtube fragment for live updates
    -Countdown Timer
    -Weather Forecast for the day
    -Calendar(I've seen this issue started somewhere)
If there is a way i can go about adding any of the above to slides, please let me know.

**Comment by Amuri-Bonface** [comment]
Anyone around?

**Comment by eerotal** [comment]
Sure there's a way to do that. It'll require some programming though.

The markup tags are defined in [src/node_modules/ls-markup/eval_defs.js](https://github.com/eerotal/LibreSignage/blob/master/src/node_modules/ls-markup/eval_defs.js). It's quite simple to implement new tags just by replicating the existing ones in that file. You'll have to write all the code that makes the markup tags work yourself though and that might take some time.

A calendar markup tag is indeed something I need to look into but I haven't had the time yet.

**Comment by Amuri-Bonface** [comment]
Thanks for the update. I really need those specific tags I've outlined. If you have time try and implement some. In the meantime let me see if i can sort one or two tags.

**Comment by eerotal** [comment]
I will eventually work on them but it's unfortunately not my top priority right now. I have another big change I'm working on and I'm also rather busy right now.

**Comment by Qualitymix** [comment]
This is of particular interest to me as well.  I'm looking at implementing this software in our small school, and the weather functionality is something that has been requested.  I'll take a look at the source, but I'm not sure there is too much I can do. NodeJS/NPM isn't my strong suit, and I don't have the time to dedicate to picking up another language right now!

**Comment by eerotal** [comment]
@Qualitymix I'll get around to implementing it at some point but cleanly integrating something like this into LibreSignage is not really a trivial thing to do. I've also been really busy with other things recently so I haven't had much time to work on LibreSignage.

**Comment by Qualitymix** [comment]
Taking a look at how the tags work, they are just implementing HTML code, which is awesome.  Really, if there could be a way to build a custom URL out of options passed through a tag, we could get a [weather] tag implemented very easily.  Using a site like https://weatherwidget.io/ would work great!

**Comment by Qualitymix** [comment]
So I can compile a hardcoded version of what I'm looking for, however I run into an issue with what seems like the javascript not liking the <script> tag too awfully much.   It works at first in the editor, but after a bit the graphic disappears, and the displays never show the content, save for some text.

Where abouts could I hardcode a script for the main HTML page?

**Comment by eerotal** [comment]
@Qualitymix  That's odd. What changes did you make to the markup transpiler code?

**Comment by Qualitymix** [comment]
All I did was add a tag in the eval_defs.js file inside ls-markup.   

`weather: {
   has: {},
   handle: () => {
      return [
         '<div....  and so on.  This part is what was returned from using the willyweather.com widget creator' </div>`,
         ``
      ];
      }
   },`

This code works with the willweather widgets.  Anythingn with the <script> tag doesn't seem to work.

**Comment by eerotal** [comment]
That's odd. I need to look into this myself sometime.

**Comment by Canonip** [comment]
How about [html] and [script] tags?
Then the user can add whatever they like.

I tried a workaround for Fontawesome: 
`[p][font f="Font Awesome 5 Brands"]&#xf1f0;[/font][/p]`
but it sanitizes the ampersand, so the hexcode will be shown

**Comment by eerotal** [comment]
I think instead of implementing HTML and script tags it'd make more sense to implement a way to create slides entirely using HTML, CSS and JS scripts. Mixing HTML and scripts with the LibreSignage markup would probably introduce all kinds of hard to debug problems with slide layouts. I've thought of implementing HTML & CSS slide editing before and it's a feature that will probably be implemented in the future.

**Comment by Qualitymix** [comment]
I have added a raw HTML tag that works okay.  It's a little wonky, and I wind up with some extra ' characters.

**Comment by eerotal** [comment]
@Qualitymix Yeah, it's certainly possible to add a HTML tag to the markup transpiler, but it's really a matter of whether it's actually a good idea to do that. Like I commented before, I'm not too keen on implementing a HTML tag in the official LibreSignage builds, but if it works for you then that's great. I need to look into implementing bare HTML & CSS slide editing in the future though.

**Comment by shark2380** [comment]
> I have added a raw HTML tag that works okay. It's a little wonky, and I wind up with some extra ' characters.

Good evening, could you share some hint on how you do that?
